### ISSUE TITLE
High consumption of RAM

### ISSUE THREAD

**Issue by arthurmteodoro** [amaiya/ktrain#139]
I am using the library in the Google Colab environment and I am having problems with the high consumption of RAM, with the notebook being shut down due to lack of memory.

I've tried to reduce the number of workers, as well as the batch size, however, the problem still remains. I am using the HAM10000 dataset with original dataset images and generated by Keras ImageGenerator, with a total of 38569 for training. I'm using the pre-trained ResNet50 with the imagenet dataset.

**Comment by amaiya** [comment]
Hello @arthurmteodoro   I wasn't able to reproduce this issue.  

Are you running out of GPU memory or system memory?  I did a quick test on HAM10000 dataset using a TITAN V GPU and had no problems with system or GPU memory. In fact, it trained pretty fast.  I've trained ResNet50 image classifiers on Google Colab with other datasets and also did not have any problems.   Here are some things you can try:

- The original dataset has 10,000 images.  I'm assuming  you have 38,569 images because you did offline data augmentation?  Perhaps, these images are causing a problem.  Try on-the-fly data augmentation as described in the tutorials and examples and see if that helps.
- You can try a model with less parameters like MobileNet, although ResNet50 should work fine on Google Colab under normal circumstances.
-  I've seen several reports of memory leaks in TF 2.0 and TF 2.1 on the TensorFlow GitHub repo. I myself have seen such issues related to image generators in TF 2.0, but not in TF 2.1.0 for some reason..  But others have told me they've seen such issues in TF 2.1.0 as well.  Although the current version of *ktrain* is tested only on TF 2.1.0, you can try installing a different version of TF 2.x like TF 2.2.0-rc3 (which is the default version on Google Colab for some reason).  Note that you must install TF **after** *ktrain* is installed as `pip3 install ktrain` will install TF 2.1.0 and remove existing installations.  Also note that, due to an undocumented breaking change in TF 2.2.0-rc3, `view_top_losses` will not currently work in TF 2.2.0.  This will be fixed once TF 2.2.0 is officially released.  As of right now, TF 2.1.0 is the current version and the current version of *ktrain* is implemented to work with that version of TensorFlow. Most everything else should work in TF 2.2.0-rc3 right now, though.

**Comment by arthurmteodoro** [comment]
Hello,

I tested again with the same parameters and input data and I didn't have the same memory overflow as before. Even so, I did the test using on-the-fly data augmentation and I also didn't get the same memory overflow. Besides, I did the training with the ResNet38 network, using the offline data, and got the same result, all right.

I believe it was an error or in TensorFlow, which may appear sporadically, or an error in my code.

Thank you!
